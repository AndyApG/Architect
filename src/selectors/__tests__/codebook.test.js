/* eslint-env jest */

import testState from '../../__tests__/testState.json';
import {
  getVariableOptionsForSubject,
} from '../codebook';

describe('codebook selectors', () => {
  describe('getVariableOptionsForSubject()', () => {
    it('extracts variables for nodeType into options list for node entity', () => {
      const subject = {
        type: 'bar',
        entity: 'node',
      };

      const result = getVariableOptionsForSubject(
        testState,
        subject,
      );

      expect(result).toMatchSnapshot();
    });

    it('extracts variables for nodeType into options list for ego entity', () => {
      const subject = {
        type: undefined,
        entity: 'ego',
      };

      const result = getVariableOptionsForSubject(
        testState,
        subject,
      );

      expect(result).toMatchSnapshot();
    });
  });
});
