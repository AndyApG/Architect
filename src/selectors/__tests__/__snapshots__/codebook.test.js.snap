// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codebook selectors getVariableOptionsForSubject() extracts variables for nodeType into options list for ego entity 1`] = `
Array [
  Object {
    "label": "SIERRA",
    "type": "text",
    "value": "sierra",
  },
  Object {
    "label": "TANGO",
    "type": "text",
    "value": "tango",
  },
]
`;

exports[`codebook selectors getVariableOptionsForSubject() extracts variables for nodeType into options list for node entity 1`] = `
Array [
  Object {
    "label": "ALPHA",
    "type": "text",
    "value": "alpha",
  },
  Object {
    "label": "BRAVO",
    "type": "text",
    "value": "bravo",
  },
  Object {
    "label": "CHARLIE",
    "type": "location",
    "value": "charlie",
  },
]
`;
